crates/services/block_aggregator_api/src/blocks/importer_and_db_source/serializer_adapter.rs

```diff
@@ -163,36 +249,1276 @@ fn proto_tx_from_tx(tx: FuelTransaction) -> ProtoTransaction {
+    contract: &fuel_core_types::fuel_tx::output::contract::Contract,
+) -> ProtoContractOutput {
+    ProtoContractOutput {
+        input_index: u32::from(contract.input_index),
+        balance_root: bytes32_to_vec(&contract.balance_root),
+        state_root: bytes32_to_vec(&contract.state_root),
+    }
+}
+
+fn proto_output_from_output(output: Output) -> ProtoOutput {
+    let variant = match output {
+        Output::Coin {
+            to,
+            amount,
+            asset_id,
+        } => ProtoOutputVariant::Coin(ProtoCoinOutput {
+            to: to.as_ref().to_vec(),
+            amount,
+            asset_id: asset_id.as_ref().to_vec(),
+        }),
+        Output::Contract(contract) => {
+            ProtoOutputVariant::Contract(proto_contract_output_from_contract(&contract))
+        }
+        Output::Change {
+            to,
+            amount,
+            asset_id,
+        } => ProtoOutputVariant::Change(ProtoChangeOutput {
+            to: to.as_ref().to_vec(),
+            amount,
+            asset_id: asset_id.as_ref().to_vec(),
+        }),
+        Output::Variable {
+            to,
+            amount,
+            asset_id,
+        } => ProtoOutputVariant::Variable(ProtoVariableOutput {
+            to: to.as_ref().to_vec(),
+            amount,
+            asset_id: asset_id.as_ref().to_vec(),
+        }),
+        Output::ContractCreated {
+            contract_id,
+            state_root,
+        } => ProtoOutputVariant::ContractCreated(ProtoContractCreatedOutput {
+            contract_id: contract_id.as_ref().to_vec(),
+            state_root: bytes32_to_vec(&state_root),
+        }),
+    };
+
+    ProtoOutput {
+        variant: Some(variant),
+    }
+}
+
+fn proto_upgrade_purpose(purpose: &UpgradePurpose) -> ProtoUpgradePurpose {
+    let variant = match purpose {
+        UpgradePurpose::ConsensusParameters {
+            witness_index,
+            checksum,
+        } => ProtoUpgradePurposeVariant::ConsensusParameters(
+            ProtoUpgradeConsensusParameters {
+                witness_index: u32::from(*witness_index),
+                checksum: checksum.as_ref().to_vec(),
+            },
+        ),
+        UpgradePurpose::StateTransition { root } => {
+            ProtoUpgradePurposeVariant::StateTransition(ProtoUpgradeStateTransition {
+                root: root.as_ref().to_vec(),
+            })
+        }
+    };
+
+    ProtoUpgradePurpose {
+        variant: Some(variant),
     }
 }
 
 fn proto_policies_from_policies(
     policies: &fuel_core_types::fuel_tx::policies::Policies,
 ) -> ProtoPolicies {
-    const POLICY_ORDER: [PolicyType; 5] = [
-        PolicyType::Tip,
-        PolicyType::WitnessLimit,
-        PolicyType::Maturity,
-        PolicyType::MaxFee,
-        PolicyType::Expiration,
-    ];
-
-    let values = POLICY_ORDER
-        .iter()
-        .map(|policy_type| policies.get(*policy_type).unwrap_or_default())
-        .collect();
-
+    let mut values = [0u64; 6];
```

Comment:
Why values for policies in an empty array? If the policy is not set, we shouldn't include them. If no policy is set, then it will be empty vector.